# CoCo Dataset #

## General ##

Number of images: 118287 

### Embeddings ###

|model|JIT|distribution|read Duration|inference_duration|write_duration|total_duration|sample_count|size|
|---|---|---|---|---|---|---|---|---|
|ViT-B/32|yes|sequential|37.42|164.62|0.02|202.06(3.3 min)|118287|121126016 (116 Mb)|
|ViT-B/32|no|sequential|15.87|186.23|0.02|202.13|118287|121126016|
|ViT-B/32|yes|pyspark|32.67|166.09|0.03|198.80|118287|121126016|
|ViT-L/64|yes|sequential|6.10|3220.61|0.05|3226.76(53.8 min)|118287|181688960 (174 Mb)|
|ViT-L/64|no|sequential|5.85|3392.02|0.05|3397.92|118287|181688960|
|ViT-L/64|yes|pyspark|6.07|3220.89|0.05|3227.01|118287|181688960|


Estimated effort for 400m:

Size: 573 Gb

In [7]:
import os, glob, json
output_folder= "/home/junjie/clip_test/index_coco_train2017_pyapi_pyspark"
jfile = glob.glob(output_folder+"/*.json")
with open(jfile[0]) as stats_file:
    stats = json.load(stats_file)
    print(stats)

{'index_key': 'HNSW32', 'index_param': 'efSearch=915', 'index_path': '/home/junjie/clip_test/index_coco_train2017_pyapi_pyspark/image.index', 'size in bytes': 274453014, 'avg_search_speed_ms': 9.525284638539667, '99p_search_speed_ms': 23.022338580340143, 'reconstruction error %': 0.0, 'nb vectors': 118287, 'vectors dimension': 512, 'compression ratio': 0.8826712174492644}


### Index ###

|model|JIT|distribution|size_in_bytes|avg_search_speed_ms|99p_search_speed_ms|nb_vectors|vectors_dimension|compression_ratio|
|---|---|---|---|---|---|---|---|---|
|ViT-B/32|yes|sequential|274453014(262 Mb)|9.27|19.99|118287|512|0.88|
|ViT-B/32|no|sequential|274453014|9.91|19.60|118287|512|0.88|
|ViT-B/32|yes|pyspark|274453014|9.52|23.02|118287|512|0.88|
|ViT-L/64|yes|sequential|395578902 (377 Mb)|9.21|18.73|118287|768|0.92|
|ViT-L/64|no|sequential|395578902|9.45|19.78|118287|768|0.92|
|ViT-L/64|yes|pyspark|395578902|10.76|32.66|118287|768|0.92|


Estimated effort for 400m:

Size: 1.22 Tb